include/lokinet/lokinet_context.h

```diff
@@ -13,10 +14,16 @@ extern "C"
 
   struct lokinet_context;
 
-  /// allocate a new lokinet context
-  struct lokinet_context* EXPORT
+  struct lokinet_config* EXPORT
   lokinet_context_new();
 
+  /// allocate a new lokinet context, passing in a loaded config, finalizing the config's loaded
+  /// values and injecting them into the lokinet context. if config is NULL we will load sensible
+  /// defaults. after calling lokinet_context_new the config can be released by calling
+  /// lokinet_config_free().
+  struct lokinet_context* EXPORT
+  lokinet_context_from_config(struct lokinet_config*);
```

Comment:
I think there should also be a convenience method taking a filename (to do the load + initialization from the file) for the likely most common use case.